src/main/proxy-functions/shell-api-request.ts

```diff
@@ -19,29 +19,38 @@
  * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 
-import type http from "http";
 import url from "url";
 import logger from "../logger";
-import * as WebSocket from "ws";
+import { Server as WebSocketServer } from "ws";
 import { NodeShellSession, LocalShellSession } from "../shell-session";
 import type { ProxyApiRequestArgs } from "./types";
 import { ClusterManager } from "../cluster-manager";
+import { appName, appSemVer } from "../../common/vars";
+import { LensProxy } from "../lens-proxy";
 
-export function shellApiRequest({ req, socket, head }: ProxyApiRequestArgs) {
-  const ws = new WebSocket.Server({ noServer: true });
+export function shellApiRequest({ req, socket, head }: ProxyApiRequestArgs): void {
+  const cluster = ClusterManager.getInstance().getClusterForRequest(req);
 
-  ws.on("connection", ((socket: WebSocket, req: http.IncomingMessage) => {
-    const cluster = ClusterManager.getInstance().getClusterForRequest(req);
+  if (
+    socket.remoteAddress !== "127.0.0.1"
+    || !req.headers["user-agent"]?.includes(` ${appName}/${appSemVer} `)
+    || !req.headers.origin?.endsWith(`.localhost:${LensProxy.getInstance().port}`)
+    || !cluster
+  ) {
+    socket.write("Invalid shell request");
```

Comment:
can this throw?

Reply:
No, the documentation states that this function will return `true` if all of the buffer (or string in this case) has been flushed to kernel memory or `false` if at least some of it was buffered in user memory. 

And then it goes on to talk about not writing more to a socket after `.write()` returns false until the `"drain"` event is emitted for other security reasons, but since I am only writing once I don't think that matters.